### PR TITLE
Make Parameter dumb again

### PR DESCRIPTION
**Pull request by labkey-matthewb**
#### Rationale
One Parameter constructor was substituting getJdbcRsName() instead of using the passed in column name. This breaks code since the caller expects to use the column name.

https://www.labkey.org/home/Developer/issues/issues-details.view?issueId=45368

The getJdbcRsName() code was a "fix" for https://www.labkey.org/home/Developer/issues/issues-details.view?issueId=26694.  Hopefully we have regression tests for that.

#### Related Pull Requests
* 

#### Changes
* removed getJdbcRsName() code
* removed Parameter members that it didn't use itself (scale, parameter) and changed caller to keep track of these.

### CODE REVIEW

File: api/src/org/labkey/api/data/Parameter.java [hunk 1]
```diff
@@ -261,13 +232,11 @@ public void setValue(@Nullable Object in)
                 return;
             }
 
-            if (value instanceof AttachmentFile)
+            if (value instanceof final AttachmentFile attachmentFile)
```
Comment (labkey-jeckels) [inline api/src/org/labkey/api/data/Parameter.java:235]: I didn't know this was a supported syntax!

**Review by labkey-jeckels — APPROVED** [review]

**Comment by labkey-bpatel** [comment]
@labkey-matthewb can we please merge to release22.7-SNAPSHOT

File: api/src/org/labkey/api/data/Parameter.java (hunk 1 above)
Comment (labkey-bpatel) [reply api/src/org/labkey/api/data/Parameter.java:235]: yeah ditto, had to do a double take on this line
pretty cool

**Comment by labkey-matthewb** [comment]
> @labkey-matthewb can we please merge to release22.7-SNAPSHOT

I'd rather not?  Seems like a bit much for a stable release.  But I'll defer to triage/Josh.

**Comment by labkey-bpatel** [comment]
> > @labkey-matthewb can we please merge to release22.7-SNAPSHOT
> 
> I'd rather not? Seems like a bit much for a stable release. But I'll defer to triage/Josh.

@labkey-matthewb i didn't see Hannah's comment on the [ticket](https://www.labkey.org/home/Developer/issues/issues-details.view?issueId=45368), it is targeted for 22.11, so nevermind my comment

**Pull request merged**